src/storage/v2/indices/text_index.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "storage/v2/indices/text_index.hpp"
+#include "query/db_accessor.hpp"
+#include "storage/v2/view.hpp"
+#include "text_search.hpp"
+
+namespace memgraph::storage {
+
+void TextIndex::AddNode(Vertex *vertex_after_update, Storage *storage,
+                        const std::vector<mgcxx::text_search::Context *> &applicable_text_indices) {
+  // NOTE: Text indexes are presently all-property indices. If we allow text indexes restricted to specific properties,
+  // an indexable document should be created for each applicable index.
+  nlohmann::json document = {};
+  nlohmann::json properties = {};
+  for (const auto &[prop_id, prop_value] : vertex_after_update->properties.Properties()) {
+    if (!prop_value.IsString()) continue;
```

Comment:
In general, here shouldn't be only string because other data types could also be searched

Reply:
Added more types